metrics/framework/Metrics.hpp

```diff
@@ -0,0 +1,305 @@
+/*##############################################################################
+    HPCC SYSTEMS software Copyright (C) 2021 HPCC Systems®.
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+       http://www.apache.org/licenses/LICENSE-2.0
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+############################################################################## */
+
+#pragma once
+
+#include <string>
+#include <utility>
+#include <vector>
+#include <map>
+#include <atomic>
+#include <chrono>
+#include <mutex>
+#include <memory>
+#include <unordered_set>
+#include "jiface.hpp"
+#include "jptree.hpp"
+
+#ifdef METRICSDLL
+#define METRICS_API DECL_EXPORT
+#else
+#define METRICS_API DECL_IMPORT
+#endif
+
+namespace hpccMetrics {
+
+/*
+ * Enumerates the type of the value in a measurement
+ */
+enum ValueType {
```

Comment:
The purpose for the value type is in case a sink has to deserialize a value reported by a metric. Or, should it need to add a suffix to a reported metric name. The type was added after discussions with Rigo in the earlier stages of the framework development.

Reply:
The types the framework reports are determined by the metrics we want to report from our platform.  There should need to be a strong reason why we need anything other than integer.